src/develop/format.c

```diff
@@ -79,21 +59,38 @@ void default_input_format(dt_iop_module_t *self, dt_dev_pixelpipe_t *pipe, dt_de
 void default_output_format(dt_iop_module_t *self, dt_dev_pixelpipe_t *pipe, dt_dev_pixelpipe_iop_t *piece,
                            dt_iop_buffer_dsc_t *dsc)
 {
-  _get_iop_priorities(self);
-
   dsc->channels = 4;
   dsc->datatype = TYPE_FLOAT;
+  dsc->cst = self->output_colorspace(self, pipe, piece);
 
-  if(self->priority >= _iop_module_demosaic) return;
+  if(dsc->cst != iop_cs_RAW) return;
 
   if(pipe->image.flags & DT_IMAGE_RAW) dsc->channels = 1;
 
-  if(self->priority >= _iop_module_rawprepare) return;
+  if(dt_ioppr_get_iop_order(pipe->iop_order_list, self->op) >= dt_ioppr_get_iop_order(pipe->iop_order_list, "rawprepare")) return;
 
   if(piece->pipe->dsc.filters)
     dsc->datatype = TYPE_UINT16;
 }
 
+int default_input_colorspace(dt_iop_module_t *self, dt_dev_pixelpipe_t *pipe,
+                     dt_dev_pixelpipe_iop_t *piece)
+{
+  return self->default_colorspace(self, pipe, piece);
```

Comment:
is that expected that this is self->default_coulorspace and not self->input_colorspace().
likewise for the two others implementations below (->blend_colorspace() and ->output_colorspace())

Reply:
default_colorspace is mandatory, the other are optional and the default_colorspace is the default (if they are not implemented)